ViAn/Filehandler/filehandler.h

```diff
@@ -15,93 +15,107 @@
     void save_project(Project* proj);
-    // Video operations
+    bool save_project(Project* proj, ID dir_id, FileHandler::SaveFormat save_format);
+
+//  Video methods
     void remove_video_from_project(ID proj_id, ID vid_id);
     ID add_video(Project* proj, std::string file_path);
-    //directory manipulation
-    //varying implementation
-    ID create_directory(std::string dir_path);
-    FH_ERROR delete_directory(ID id);
+//  Directory methods
+    ID create_directory(QString dir_path);
+    bool delete_directory(ID id);
 
-    //file manipulation
+//  File methods
 
-    ID create_file(std::string file_name, ID dir_id);
-    FH_ERROR delete_file(ID id);
-    void write_file(ID id, std::string text, WRITE_OPTION opt = WRITE_OPTION::APPEND);
-    void read_file(ID id,  std::string& buf, int lines_to_read = -1);
+    ID create_file(QString file_name, QDir dir);
+    bool delete_file(ID id);
+    void write_file(ID id, QString text, WRITE_OPTION opt = WRITE_OPTION::APPEND);
+    void read_file(ID id, QString& buf, int lines_to_read = -1);
 
-    friend bool operator==(ProjFiles& pf, ProjFiles& pf2);
     friend bool operator==(Project& proj, Project& proj2);
 
     bool proj_equals(Project& proj, Project& proj2);
-    bool projfiles_equal(ProjFiles& pf, ProjFiles& pf2);
     bool dirs_equal(ID id, ID id2);
     bool files_equal(ID id, ID id2);
     // thread safe read operations for maps
 
 
-    std::string get_dir(ID id);
+    QDir get_dir(ID id);
     Project* get_project(ID id);
-    std::string get_file(ID id);
+    QString get_file(ID id);
 
     // Last error
-    FH_ERROR last_error;
+    bool last_error;
 
 private:
 
-    void update_proj_files(Project* proj); // used to update existing project files and maps
+    void update_proj_file(Project* proj); // used to update existing project files and maps
     // thread safe add operations for maps
-    ID add_file(std::string file_path);
-    void add_project(std::pair<ID,Project*> pair);
+    ID add_file(QString file);
+    ID add_project(Project* proj);
+    void add_project(ID id, Project *proj);
 
-    ID add_dir(std::string dir_path);
+    ID add_dir(QDir dir);
+    void add_dir(ID dir_id, QDir dir);
     ID load_project_file(std::string file_path, std::stringstream& proj_file_stream);
     void load_proj_files(std::string str);
     //add used for loading project from file
-    void add_file(ID id , std::string file_path);
+    void add_file(ID id , QString file);
 
     /**
      * @brief m_projects, m_fileMap, m_dirMap
      * map structures for keeping track of projects, files and directories.
      */
     std::map<ID,Project*> projects;
-    std::map<ID, std::string> file_map;
-    std::map<ID, std::string> dir_map;
+    std::map<ID, QString> file_map;
+    std::map<ID, QDir> dir_map;
     /**
      * @todo implement smarter lock mechanism to avoid overhead
      * of only 1 reader/writer at a time
      * @brief dirMapLock, fileMapLock, projMapLock
```

Comment:
Same here change the comment to match the variable names. Also please have an empty line before the comment, between line 107/108.